### DISCUSSION TITLE
JSON response to newly created Record

### DISCUSSION THREAD

**Discussion by dankeyy** [pocketbase/pocketbase#2790] [Q&A] [answered]
Hi, 
I'm making a POST request to create a file record, processing it in the backend (using a subprocess, if it matters), eventually outputting a json file with some contents. I want to respond to the POST request with this json.

onRecordAfterCreateRequest is great, because it gives you a handle to the IDs of the collection and file.
But it's not exactly what I want because you can't really return from it, it's just a hook.

app.OnBeforeServe is great, because it allows you to return a custom json. 
But it's not exactly what I want because then I don't have a pointer to the newly created Record.

Making the json could take a bit of time, so I can't make a new record for the json internally then make a GET request for it, that'd be a race, and also kinda hacky, also no real need for 2 requests ideally.

Is there a way to get the utility of the above two functions together?
And sorry if I missed something in the docs about it, but I read it a couple times and couldn't come up with a satisfying solution. 
Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not entirely sure that I understand what is the requirement and what exactly do you mean by "_processing it in the backend (using a subprocess, if it matters)_", but with the upcoming v0.17.0 release you'll be able to write a response body in a `OnRecordAfter*Request` hook, if that's what you are asking.



Currently, the only way I can think this could be done is to attach in a `OnRecordBefore*Request` hook and add your custom content as unknown dynamic Record field, something like:

```go
app.OnRecordBeforeCreateRequest("yourCollection").Add(func(e *core.RecordCreateEvent) error {
	admin, _ := e.HttpContext.Get(apis.ContextAdminKey).(*models.Admin)
	if admin != nil {
		return nil // ignore admin requests
	}

	e.Record.WithUnknownData(true)
	e.Record.Set("someCustomField", "anything")

	return nil
})
```

**Reply by dankeyy** [reply]
by "it" I mean the file posted. I POST a text file, make some filtering/ changes (based on certain logic) on the text file data, and output a json containing that data. I want to return this json data as a response to the request which I'm handling right now.

OnRecordAfter*Request sounds nice, when will v0.17.0 be out?

And yes so let me get your current solution straight . I would access the current posted file via e.Record, add a field for the json data I wish to return, then GET request the Record I just posted and grab the json field?

**Reply by ganigeorgiev** [reply]
I'm sorry, I'm confused. There is no need to send a separate GET request. The above "someCustomField" will be returned as part of the record in the create response.

If that's not what you are looking for, please elaborate what request you are sending, what operations you are performing and what is the expected response.

**Reply by dankeyy** [reply]
Oh right it's in there. Well this was helpful,  thanks.
